### ISSUE TITLE
Add ChordPro to Windows system path

### ISSUE THREAD

**Issue by AnniePerson** [ChordPro/chordpro#114]
I installed ChordPro 0.975 on Win10 into the standard location c:\Program Files (x86)\ChordPro.ORG\ChordPro\chordpro.exe
To be able to use the CLI of ChordPro.exe conveniently from other locations, I added  c:\Program Files (x86)\ChordPro.ORG\ to the system path in the Win10 environment variables. That should be better done automatically by the installer.
Or is there another way to do that?

**Comment by sciurius** [comment]
I agree this should be an installer function. However, much to my surprise, InnoSetup has no support for adding the application path to the system path decently, and, also important, remove it when uninstalled.
There are some workarounds, each with pros and cons, that I would rather avoid.

I'll leave the issue as enhancement.

**Comment by jashearwood** [comment]
I too installed Chordpro into the standard location standard location c:\Program Files (x86)\ChordPro.ORG\ChordPro\

However, the shortcut to the gui has the path as c:\Program Files (x86)\ChordPro.ORG\ChordPro\bin\ so can't find the wxchordpro.exe executable. This path is in two places within the shortcut.

**Comment by sciurius** [comment]
I just verified that the installer installs the shortcut as C:\Program Files (x86)\ChordPro.ORG\ChordPro\wxchordpro.exe .

![VirtualBox_Win10Pro_26_02_2021_19_38_52](https://user-images.githubusercontent.com/1118804/109341323-8e0cc480-786a-11eb-91e8-6044f5606e33.png)

**Comment by jashearwood** [comment]
Great. But my new installation put in the path I gave earlier. Today I downloaded the installer using the link here https://www.chordpro.org/chordpro/install-windows-native/ which downloaded ChordPro-installer-0-977-0-2-msw-x64.exe. On the face of it, I should have had the same result as you. 

Are you using an updated installer? Did you update a previous version?

Although I'm a newbie to Chordpro, I'm quite experienced with computers which is how I managed to work out what was going on when Windows complained it couldn't find the target exe file. If you want screenshots, I'll uninstall and start again if you like.

**Comment by sciurius** [comment]
What installer did you use that gave the wrong results?

Have you tried uninstalling ChordPro, then install fresh using the installer from GitHub?

**Comment by jashearwood** [comment]
The page I downloaded it from is here: https://github.com/ChordPro/chordpro/releases/tag/R0_977 This is linked to by the page I mentioned in the last post.

and, yes, I installed it yesterday, had the problem and uninstalled without investigating too much. Decided today to try again and was more persistent in diagnosing what had gone wrong.

Have jusrt uninstalled and reinstalled again. Attached are screenshots.

One thing that occurs to me: I don't like a cluttered Desktop. A hangover from years of IT support before I moved to a more technical job.

The only shortcut that I have is the one in the Start menu. It may be that the shortcut from the Desktop is different.

![Screenshot 2021-02-26 22 02 55](https://user-images.githubusercontent.com/79711224/109360275-d209c480-787e-11eb-92d4-ef75966ff6dc.png)
![Screenshot 2021-02-26 22 03 34](https://user-images.githubusercontent.com/79711224/109360276-d2a25b00-787e-11eb-817a-b6e7ac418dea.png)

Sorry to be a pain!

**Comment by sciurius** [comment]
This is very surprising. The desktop shortcut is installed okay, but the entry in the start menu is wrong.

I opened a issue: https://github.com/ChordPro/chordpro/issues/135

**Comment by jashearwood** [comment]
Thanks.

I should have said which shortcut was incorrect in the first place but, presumably like yourself, I didn't think there would be a difference between the startup and the desktop shortcut.

**Comment by sciurius** [comment]
Please try  ChordPro-installer-0-977-0-3-msw-x64.exe .

**Comment by jashearwood** [comment]
That version works fine.

**Comment by nomike** [comment]
BTW: Rather than adding the ChordPro installation directory to the path, it might be worth to look at this registry feature:
https://docs.microsoft.com/en-us/windows/win32/shell/app-registration

This is what allows you to put "winword" in Start --> Run to run Microsoft Word.

**Comment by sciurius** [comment]
This should be handled by the installer program. AFAIK InnoSetup can not do this (at least I haven't found a way).
